src/client/runner/api/actions.js

```diff
@@ -148,24 +148,14 @@ function ensureElementVisibility (element, actionName, callback) {
     var success = false;
 
     if (domUtils.isOptionElement(element) || domUtils.getTagName(element) === 'optgroup') {
-        var parentSelect = domUtils.getSelectParent(element);
-
-        if (!parentSelect) {
+        if (selectElement.isOptionElementVisible(element))
```

Comment:
BTW, do we have tests for the legacy API to ensure that we didn't break anything?